apps/wallet-dashboard/lib/utils/migration.ts

```diff
@@ -50,52 +67,236 @@ export function groupStardustObjectsByMigrationStatus(
+            }
+        }
+    }
+
+    return flatObjects;
+}
+
+async function getNftDetails(
+    object: IotaObjectData,
+    expirationKey: UnlockConditionTimestamp,
+    client: IotaClient,
+): Promise<ResolvedNftObject[]> {
+    const objectFields = extractMigrationOutputFields(object);
+    const nftOutputDynamicFields = await client.getDynamicFields({
+        parentId: objectFields.id.id,
+    });
+
+    const nftDetails: ResolvedNftObject[] = [];
+    for (const nft of nftOutputDynamicFields.data) {
+        const nftObject = await client.getObject({
+            id: nft.objectId,
+            options: { showDisplay: true },
+        });
+
+        if (!nftObject?.data?.display?.data) {
+            continue;
+        }
+
+        nftDetails.push({
+            balance: BigInt(objectFields.balance),
+            name: nftObject.data.display.data.name ?? '',
+            image_url: nftObject.data.display.data.image_url ?? '',
+            commonObjectType: CommonMigrationObjectType.Nft,
+            unlockConditionTimestamp: expirationKey,
+            output: object,
+            uniqueId: nftObject.data.objectId,
+        });
+    }
+
+    return nftDetails;
+}
+
+async function extractNativeTokensFromObject(
+    object: IotaObjectData,
+    client: IotaClient,
+    expirationKey: UnlockConditionTimestamp,
+): Promise<ResolvedNativeToken[]> {
+    const fields = extractMigrationOutputFields(object);
+    const bagId = fields.native_tokens.fields.id.id;
+    const bagSize = Number(fields.native_tokens.fields.size);
+
+    const nativeTokens = bagSize > 0 ? await getNativeTokensFromBag(bagId, client) : [];
+    const result: ResolvedNativeToken[] = [];
+
+    for (const nativeToken of nativeTokens) {
+        const nativeTokenParentId = fields.native_tokens.fields.id.id;
+        const objectDynamic = await client.getDynamicFieldObject({
+            parentId: nativeTokenParentId,
+            name: nativeToken.name,
+        });
+
+        if (objectDynamic?.data?.content && 'fields' in objectDynamic.data.content) {
+            const nativeTokenFields = objectDynamic.data.content.fields as {
+                name: string;
+                value: string;
+                id: { id: string };
+            };
+            const tokenStruct = extractObjectTypeStruct(nativeTokenFields.name);
+            const tokenName = tokenStruct[2];
+            const balance = BigInt(nativeTokenFields.value);
+
+            result.push({
+                name: tokenName,
+                balance,
+                coinType: nativeTokenFields.name,
+                unlockConditionTimestamp: expirationKey,
+                commonObjectType: CommonMigrationObjectType.NativeToken,
+                output: object,
+                uniqueId: nativeTokenFields.id.id,
+            });
+        }
+    }
+
+    return result;
+}
+
+function getGroupingObjectsKey(
+    isTimelockUnlockCondition: boolean,
+    objectFields: CommonOutputObjectWithUc,
```

Comment:
This function seems strange to me, an object of type `CommonOutputObjectWithUc` is passed, and if it does not have uc, this key `MIGRATION_OBJECT_WITHOUT_UC_KEY` is set
how is this possible?